service/controller/resource/ipam/create.go

```diff
@@ -24,7 +24,12 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "acquiring lock for IPAM")
 		err := r.locker.Lock(ctx)
 		if locker.IsAlreadyExists(err) {
+			// In case the lock already exists we stop here and try again during
+			// the next reconciliation loop because another process is already
+			// trying to allocate subnets.
 			r.logger.LogCtx(ctx, "level", "debug", "message", "lock for IPAM is already acquired")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			return nil
```

Comment:
@tuommaki can you please have a look if this makes sense to you?

Reply:
I think we need to cancel the whole loop as without the subnets we should not continue with next resource

Reply:
When the subnets are not allocated then the handlers that expect the subnet to be allocated should check that and cancel themselves. There might be handlers that wish to do work because they do not have that dependency.

Reply:
Yeah, you are right, it would even cancel resources that already have allocated subnets which si not what we want.

Reply:
This seems to be safe, but I'm not sure if this is really needed.

Internal locking implementation of `aws-operator` uses project name & version as owner for the lock. Given that I've understood our current app / release management and associated validation rules correctly, at no time there should be more than one instance of operator running with given project name & version pair (i.e. the owner here).

`kubelock` seems to return `alreadyExistsError` in case same operator tries to acquire lock for itself again: https://github.com/giantswarm/kubelock/blob/master/lock.go#L35-L36

If owner of lock differs from currently running operator, the error would be `ownerMismatchError` and execution here should fall to next block below.

Have you had an issue with this? I feel this change is safe in the sense that lock TTL is only 30sec so in case you hit this execution branch, an additional reconciliation round is needed, but that should be it (given that there are no conflicts with other operators).

Reply:
We seem to have an issue. Notice that we run the ` ipam` handler in two controllers. One for  the masters and one for the workers. So IMO this is a race right there. When the cluster controller locks and starts working and the machine deployment controller starts looking it sees that the lock is acquired and happily continues to try the same thing.

Reply:
> Notice that we run the ipam handler in two controllers. One for the masters and one for the workers. So IMO this is a race right there. 

Oh, I didn't realize this. Given the design, wouldn't the right thing be naming the owner then accordingly per-controller?

Reply:
That would mean that you can have conflicts between the controllers and you start allocations for already taken subnets.